### PR TITLE
[SW-89] Update  SDK to v12 (typescript version)

### PR DESCRIPTION
**Pull request by subhod-i**

### CODE REVIEW

File: src/popup/components/TransferReview.vue [hunk 1]
```diff
@@ -175,12 +174,12 @@ export default defineComponent({
...
             recipient,
             hash: actionResult.hash,
-            type: 'spendToken',
+            type: 'SpendTx',
             pendingTokenTx: true,
             tx: {
               callerId: account.value.address,
               contractId: selectedAsset.contractId,
-              type: SCHEMA.TX_TYPE.contractCall,
+              type: 'ContractCallTx',
```
Comment (davidyuk) [inline src/popup/components/TransferReview.vue:182]: there is a way to convert TS Enum to string
```suggestion
              type: Tag[Tag.ContractCallTx],
```

File: src/popup/utils/constants.ts [hunk 2]
```diff
@@ -8,21 +8,27 @@ import type {
...
-  NameUpdateTx: SCHEMA.TX_TYPE.nameUpdate,
-  PayingForTx: 'payingForTx', // TODO: remove after https://github.com/aeternity/aepp-sdk-js/issues/1583 is resolved
+  spendTx: Tag.SpendTx,
+  contractCallTx: Tag.ContractCall,
+  contractCreateTx: Tag.ContractCreateTx,
+  namePreclaimTx: Tag.NamePreclaimTx,
+  nameClaimTx: Tag.NameClaimTx,
+  nameBidTx: Tag.NameClaimTx,
+  nameUpdateTx: Tag.NameTransferTx,
+  payingForTx: Tag.PayingForTx, // TODO: remove after https://github.com/aeternity/aepp-sdk-js/issues/1583 is resolved
```
Comment (davidyuk) [inline src/popup/utils/constants.ts:31]: The mentioned issues should be fixed in v13

File: src/store/plugins/sdk.js [hunk 3]
```diff
@@ -89,79 +94,78 @@ export default (store) => {
...
+          signMessage: (unsigned) => (IS_EXTENSION_BACKGROUND
+            ? sign(unsigned, store.getters.account.secretKey)
+            : store.dispatch('accounts/sign', messageToHash(unsigned))),
+          ...(IS_EXTENSION_BACKGROUND ? {} : {
+            signTransaction: (txBase64, opt) => (store.dispatch('accounts/signTransaction', { txBase64, opt })),
+          }),
+          signTransaction: (txBase64, opt) => (store.dispatch('accounts/signTransaction', { txBase64, opt })),
+        });
+
+        AeSdkWallet.prototype.getAccounts = () => ({
```
Comment (davidyuk) [inline src/store/plugins/sdk.js:118]:
```suggestion
        class MyAeSdkWallet extends AeSdkWallet {
          getAccounts() {
```
will it work to do it in OOP way?

File: src/popup/components/TransferReview.vue (hunk 1 above)
Comment (subhod-i) [reply src/popup/components/TransferReview.vue]: Done. Resolved

File: src/popup/utils/constants.ts (hunk 2 above)
Comment (subhod-i) [reply src/popup/utils/constants.ts]: Removed it.

File: src/store/plugins/sdk.js (hunk 3 above)
Comment (subhod-i) [reply src/store/plugins/sdk.js:118]: Yes. I remember. I will work on it as soon as I get time. Kept the functional approach for now.

**Review by CedrikNikita — CHANGES_REQUESTED** [review]
all the `temp` commits should be properly rebased

File: vue.config.js [hunk 4]
```diff
@@ -21,7 +21,7 @@ module.exports = {
     cordova: 'www',
     web: 'dist/web/root',
   }[PLATFORM],
-  productionSourceMap: false,
+  productionSourceMap: true,
```
Comment (CedrikNikita) [inline vue.config.js:24]: Should be switched back to `false`.

File: src/popup/utils/contracts/zeitTokenAci.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        ],
+        "kind": "contract_interface",
+        "name": "PoS",
+        "payable": false,
+        "type_defs": []
+      }
+    }
+  ],
+  "interface": "contract interface PoS =\n  stateful entrypoint set_paid : (int, int) => unit\n\nmain contract FungibleTokenFull =\n  stateful entrypoint burn_trigger_pos : (int, PoS, int) => unit\n  entrypoint balance : (address) => option(int)\n"
+}
```
Comment (CedrikNikita) [inline src/popup/utils/contracts/zeitTokenAci.js:78]: Obviously, should have a newline in the end of file.

File: src/popup/utils/constants.ts [hunk 6]
```diff
@@ -8,21 +8,26 @@ import type {
...
+export { default as PAIR_CONTRACT_ACI } from './contracts/pairContractAci';
+export { default as ZEIT_TOKEN_CONTRACT_ACI } from './contracts/zeitTokenAci';
+
 export const EXTENSION_HEIGHT = 600;
 export const MAGNITUDE = 18;
 export const SEED_LENGTH = 12;
 export const AETERNITY_CONTRACT_ID = 'aeternity';
 export const AETERNITY_SYMBOL = 'AE';
 
 export const TX_TYPE_MDW = {
```
Comment (CedrikNikita) [inline src/popup/utils/constants.ts:23]: Should be completly removed, but this can be a separated issue, depending on what is current priority.

**Comment by subhod-i** [comment]
> all the `temp` commits should be properly rebased

Done.

File: src/popup/utils/constants.ts (hunk 6 above)
Comment (subhod-i) [reply src/popup/utils/constants.ts:23]: @CedrikNikita You mean the ACIs shall be imported directly and not through constants?

File: vue.config.js (hunk 4 above)
Comment (subhod-i) [reply vue.config.js:24]: noted

File: src/popup/utils/contracts/zeitTokenAci.js (hunk 5 above)
Comment (subhod-i) [reply src/popup/utils/contracts/zeitTokenAci.js:78]: noted

File: src/popup/components/TransactionInfo.vue [hunk 7]
```diff
@@ -78,7 +78,7 @@ export default {
   },
   computed: {
     getTxType() {
-      return this.$te(`transaction.dexType.${DEX_TRANSACTION_TAGS[this.txFunction] || this.txFunction}`)
+      return this.$t(`transaction.dexType.${DEX_TRANSACTION_TAGS[this.txFunction] || this.txFunction}`)
```
[the remark this answers is not in the document]
Comment (subhod-i) [reply src/popup/components/TransactionInfo.vue:81]: noted

File: src/store/modules/accounts/index.js [hunk 8]
```diff
@@ -57,7 +57,7 @@ export default {
     },
 
     signTransaction({ getters: { active, getModule }, dispatch }, txBase64) {
-      return dispatch(`${getModule(active).name}/signTransaction`, txBase64);
+      return dispatch(`${getModule(active).name}/signTransaction`, txBase64, {});
```
Comment (CedrikNikita) [inline src/store/modules/accounts/index.js:60]: Why is that needed?

Comment (subhod-i) [reply src/store/modules/accounts/index.js:60]:
```
async signTransaction({ dispatch }, {
      txBase64,
      opt: { modal = true, host = null },
    }
  ```
  If not supplied then the default assignments of the modal and host throw error.

File: src/popup/utils/constants.ts (hunk 6 above)
Comment (CedrikNikita) [reply src/popup/utils/constants.ts:23]: Was talking about `TX_TYPE_MDW` since now they are align with types from sdk.

**Comment by Liubov-crypto** [comment]
Web wallet: 

1. Transaction/ Coin details: sort by In / Out - not working 
2. Error with switching currencies, as on the develop branch
3. Top-up invite link/claim back - failed 
4. I can't send FT even if I have them on my balance, Max function with FT also is not working 
ft send

5. Tip in tx detail is broken

tip in tx broken

After refreshing a page: 
refresh

6. The last account (7) in my wallet was not visible although there are ae tokens on balance (0.10 ae) 
7.  Notifications are loading slowly

**Comment by Liubov-crypto** [comment]
Extension: 
1. if I enter 0,1 in the amount input field it will be counted as 0 (in extension)
2. issue when changing currency 
3. Can't connect to the superhero.com
4. connection to dex - pop-up is opening in the web window

File: src/popup/utils/constants.ts (hunk 6 above)
Comment (subhod-i) [reply src/popup/utils/constants.ts]: Done

File: package.json [hunk 9]
```diff
@@ -26,16 +26,17 @@
     "release": "standard-version"
   },
   "dependencies": {
-    "@aeternity/aepp-sdk": "^11.0.1",
+    "@aeternity/aepp-sdk": "^13.0.0-alpha.0",
+    "@aeternity/aepp-sdk-11": "github:aeternity/aepp-sdk-js#tags/v11.0.1",
```
Comment (subhod-i) [inline package.json:30]: Keeping SDK 11 only for middleware swagger client generation only. Shall be removed after moving to pre-generating middleware swagger API clients

File: src/composables/maxAmount.ts [hunk 10]
```diff
@@ -19,23 +19,26 @@ import {
...
 import { IAccount } from '../types';
 import { useGetter } from './vuex';
 import { useSdk } from './sdk';
+import { FungibleContractApi } from '../types/fungibleContract';
 
 export interface UseMaxAmountOptions {
   formModel: any,
 }
 
+let tokenInstance: Contract<FungibleContractApi>;
```
Comment (subhod-i) [inline src/composables/maxAmount.ts:33]: Not a ref as `initializeContract` methods returns a reactive instance of the contract

File: src/popup/components/Modals/ConfirmTransactionSign.vue [hunk 11]
```diff
@@ -244,10 +245,11 @@ export default {
         setTimeout(() => { this.loading = false; }, 20000);
         const sdk = await getSdk();
         const { bytecode } = await sdk.getContractByteCode(this.transaction.contractId);
-        const txParams = await sdk.compilerApi.decodeCalldataBytecode({
-          bytecode,
-          calldata: this.transaction.callData,
-        });
+        const txParams = await new CompilerHttp(this.activeNetwork.compilerUrl)
```
Comment (subhod-i) [inline src/popup/components/Modals/ConfirmTransactionSign.vue:248]: Not initializing the compilerHttp at app bootup but gets initialized every time while signing a transaction. Hence, the app loads faster.  It's not often we make multiple transaction signings at once.

**Review by peronczyk — CHANGES_REQUESTED** [review]

**Comment by CedrikNikita** [comment]
When we are trying to use `CompilerHttp` from sdk
`UnsupportedVersionError: Unsupported compiler version 6.1.0. Supported: >= 7.0.1 < 8.0.0
    at Object.sendRequest (autorest.mjs?0778:91:1)
    at async Object.sendRequest (autorest.mjs?0778:66:1)
    at async CompilerHttp.sendOperationRequest
`

File: src/types/fungibleContract.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+/* eslint-disable no-unused-vars */
+export interface FungibleContractApi extends ContractMethodsBase {
+  aex9_extensions: () => [string];
+  meta_info: () => MetaInfo;
+  total_supply: () => bigint;
+  owner: () => Encoded.AccountAddress;
+  balances: () => bigint;
+  balance: (address: Encoded.AccountAddress) => Maybe<bigint>
+  transfer: (address: Encoded.AccountAddress, int: bigint) => void;
```
[the remark this answers is not in the document]
Comment (subhod-i) [reply src/types/fungibleContract.ts:27]: Keeping it as it is cause naming the common type is a bit hard and confusing.

File: src/store/plugins/sdk.js (hunk 3 above)
Comment (subhod-i) [reply src/store/plugins/sdk.js]: Done

File: src/composables/maxAmount.ts [hunk 13]
```diff
@@ -102,37 +102,34 @@ export function useMaxAmount({ store, formModel }: MaxAmountOptions) {
...
-      if (!minFee.isEqualTo(fee.value)) fee.value = minFee;
+      const spendTx: Encoded.Transaction = await sdk.value.buildTx(Tag.SpendTx, {
+        senderId: account.value.address,
+        recipientId: account.value.address,
+        amount: BigNumber(val.amount > 0 ? val.amount : 0).shiftedBy(MAGNITUDE),
+        ttl: 0,
+        nonce: nonce.value + 1,
+        payload: '',
+      } as any); // TODO: Remove typecasting to any once https://github.com/aeternity/aepp-sdk-js/issues/1727 closed.
+      const { fee } = (unpackTx(spendTx, Tag.SpendTx) as any).tx; // TODO: Remove typecasting to any once https://github.com/aeternity/aepp-sdk-js/issues/1727 closed.
```
Comment (davidyuk) [inline src/composables/maxAmount.ts:113]:
```suggestion
      const { fee } = unpackTx(spendTx, Tag.SpendTx) as any; // TODO: Remove typecasting to any once https://github.com/aeternity/aepp-sdk-js/issues/1727 closed.
```
in the latest release `unpackTx` returns just transaction properties

File: src/popup/components/Modals/ConfirmTransactionSign.vue (hunk 11 above)
Comment (davidyuk) [reply src/popup/components/Modals/ConfirmTransactionSign.vue:247]:
```suggestion
        const txParams = new CompilerHttp(this.activeNetwork.compilerUrl)
```
Compiler initialises in sync the same as node, so shouldn't be a problem to create an instance somewhere global.

File: package.json [hunk 14]
```diff
@@ -26,16 +26,17 @@
     "release": "standard-version"
   },
   "dependencies": {
-    "@aeternity/aepp-sdk": "^11.0.1",
+    "@aeternity/aepp-sdk": "^13.0.0-alpha.1",
```
Comment (davidyuk) [inline package.json:29]: I would finish this PR without FF support based on 13.0.0-alpha.0 and then update to 13.0.0-alpha.1, too many changes otherwise

File: src/store/modules/accounts/hdWallet.js [hunk 15]
```diff
@@ -44,30 +49,30 @@ export default {
...
     },
     async confirmRawDataSigning({ dispatch }, data) {
       await dispatch('modals/open', { name: MODAL_CONFIRM_RAW_SIGN, data }, { root: true });
     },
-    async confirmTxSigning({ dispatch }, { encodedTx, host }) {
+    async confirmTxSigning({ dispatch }, { txBase64, host }) {
       let txObject;
       try {
-        txObject = TxBuilder.unpackTx(encodedTx, true).tx;
+        txObject = unpackTx(txBase64).tx;
```
Comment (davidyuk) [inline src/store/modules/accounts/hdWallet.js:60]:
```suggestion
        txObject = unpackTx(txBase64);
```

File: src/store/modules/accounts/ledger.js [hunk 16]
```diff
@@ -70,24 +70,23 @@ export default {
     async signTransaction({ rootGetters: { account, 'sdkPlugin/sdk': sdk }, dispatch }, { txBase64 }) {
       await dispatch('ensureCurrentAccountAvailable');
 
-      const txObject = TxBuilder.unpackTx(txBase64).tx;
-      const binaryTx = TxBuilder.buildTx(
+      const txObject = unpackTx(txBase64).tx;
```
Comment (davidyuk) [inline src/store/modules/accounts/ledger.js:73]:
```suggestion
      const txObject = unpackTx(txBase64);
```

File: src/store/modules/accounts/ledger.js [hunk 17]
```diff
@@ -70,24 +70,23 @@ export default {
...
+      const encodedTx = buildTx(
         txObject,
-        SCHEMA.OBJECT_ID_TX_TYPE[txObject.tag],
+        Tag[txObject.tag],
         { vsn: txObject.VSN },
       ).rlpEncoded;
```
Comment (davidyuk) [inline src/store/modules/accounts/ledger.js:78]:
```suggestion
      );
```
and wrap it with `decode()`

File: src/store/modules/accounts/ledger.js [hunk 18]
```diff
@@ -70,24 +70,23 @@ export default {
...
-          binaryTx,
-          sdk.getNetworkId(),
+          encodedTx,
+          await sdk.getNetworkId(),
         ],
       }), 'hex');
 
-      return TxBuilder
-        .buildTx({ encodedTx: binaryTx, signatures: [signature] }, SCHEMA.TX_TYPE.signed).tx;
+      return buildTx({ encodedTx, signatures: [signature] }, Tag.signed).tx;
```
Comment (davidyuk) [inline src/store/modules/accounts/ledger.js:89]:
```suggestion
      return buildTx({ encodedTx, signatures: [signature] }, Tag.signed);
```

File: package.json (hunk 14 above)
Comment (subhod-i) [reply package.json:29]: Sure

File: src/store/modules/accounts/ledger.js (hunk 18 above)
Comment (subhod-i) [reply src/store/modules/accounts/ledger.js:89]: Resolving as this will be part of a separate PR

File: src/store/modules/accounts/ledger.js (hunk 16 above)
Comment (subhod-i) [reply src/store/modules/accounts/ledger.js:73]: Resolving as this will be part of a separate PR

File: src/store/modules/accounts/hdWallet.js (hunk 15 above)
Comment (subhod-i) [reply src/store/modules/accounts/hdWallet.js:60]: Resolving as this will be part of a separate PR

File: src/composables/maxAmount.ts (hunk 13 above)
Comment (subhod-i) [reply src/composables/maxAmount.ts:113]: Resolving as this will be part of a separate PR

**Comment by CedrikNikita** [comment]
There are no such thing as `Schema`, so it should be also replaced with `Tag` in the e2e test `other`.

File: src/popup/components/Modals/ConfirmTransactionSign.vue (hunk 11 above)
Comment (subhod-i) [reply src/popup/components/Modals/ConfirmTransactionSign.vue:247]: Okay.

File: src/popup/components/Modals/ConfirmTransactionSign.vue [hunk 19]
```diff
@@ -241,10 +241,9 @@ export default {
     if (this.transaction.contractId) {
       try {
         this.loading = true;
-        const { getSdk } = useSdk({ store: this.$store });
```
Comment (subhod-i) [inline src/popup/components/Modals/ConfirmTransactionSign.vue:243]: @CedrikNikita This throws an error now. Store not found.

Comment (CedrikNikita) [reply src/popup/components/Modals/ConfirmTransactionSign.vue:243]: Behaviour o this composable had changed during our work on this PR, and on rebase we were suppose to change it as well.

File: src/popup/components/Modals/ConfirmTransactionSign.vue (hunk 11 above)
Comment (subhod-i) [reply src/popup/components/Modals/ConfirmTransactionSign.vue]: Part of the sdk now.

**Comment by subhod-i** [comment]
@Liubov-crypto fixed the default name issue.

FT balances bug I observed in the develop and 1.0.2 branch too. Please confirm.

**Comment by Liubov-crypto** [comment]
> feat-sdk12.wallet.z52da5wt.xyz



> @Liubov-crypto fixed the default name issue.
> 
> FT balances bug I observed in the develop and 1.0.2 branch too. Please confirm.

The default name issue has been fixed. 

FT balances issue is also reproducible on develop branch.

File: src/composables/sdk.ts [hunk 20]
```diff
@@ -15,26 +14,16 @@ interface UseSdkOptions {
...
   const isNodeError = computed<boolean>(() => store.state.sdkPlugin.isNodeError);
 
-  /**
-   * Get the SDK instance. For now the SDK state is asynchronous.
-   * TODO: With the new SDK version this probably could be replaced with a computed prop.
-   */
-  async function getSdk(): Promise<ISdk> {
-    return watchUntilTruthy(() => sdk.value);
-  }
-
```
Comment (peronczyk) [inline src/composables/sdk.ts:32]: I suggest to not remove this function because when I was creating it the goal for me was to lazy initialize the SDK only when it's needed. So instead of the constant need of checking if the SDK is ready on the components level, this function would take care of this.
